packages/providers/src/provider.ts

```diff
@@ -205,28 +207,75 @@ export type ProviderCallParams = {
   utxoValidation?: boolean;
 };
 
+/**
+ * URL - Consensus Params mapping.
+ */
+type ConsensusParamsCache = Record<string, ConsensusParameters>;
+
 /**
  * A provider for connecting to a node
  */
 export default class Provider {
```

Comment:
I see the provider has a `processNodeInfoAndConsensusParameters` 

it should be split in 2 functions that do its part completely and independently:

`processNodeInfo`
`processChain`

then will be used to cache the correct data

Reply:
would also be nice to add a `getGasConfig` helper method in provider, that would be used like:
```
const provider = await Provider.create(url);
const gasConfig = provider.getGasConfig();

// now use correct config to create things with correct default gasConfig
const scriptRequest = new ScriptTransactionRequest({ gasPrice: gasConfig.minGasPrice, gasLimit: maxGasPerTx });
```

implementation like: 
```
getGasConfig() {
  // get cached info
  const chain = this.getChain();
  const nodeInfo = this.getNodeInfo();

  const { minGasPrice } = nodeInfo;
  const { maxGasPerTx, maxGasPerPredicate, gasPriceFactor, gasPerByte } = chain;

  // return relevant gas info
  return {
    minGasPrice, maxGasPerTx, maxGasPerPredicate, gasPriceFactor, gasPerByte
  }
}
```

Reply:
@LuizAsFight this is a great suggestion and I will get this delivered to you ASAP. But I do think we can create a separate PR for this? Does that sound OK?

Reply:
Follow-ups: #1259, #1260